ads_common/lib/ads_common/auth/oauth2_handler.rb

```diff
@@ -91,7 +91,10 @@ def get_token(credentials = nil, force_refresh = false)
 
       # Refreshes access token from refresh token.
       def refresh_token!()
-        return nil if @token.nil? or @token[:refresh_token].nil?
+        if @token.nil? or @token[:refresh_token].nil?
+          raise AuthError.new("Unable to refresh token due to missing oauth information.  Check account settings.")
```

Comment:
Any specific reasons for raising here?
I was thinking that maybe it's better to also raise in `get_token` just before returning the token value since `refresh_token!` is called conditionally. 

Even `token_from_client` has similar return logic. Could potentially add a raise there as well:
`return nil if client.refresh_token.nil? && client.access_token.nil?`

Might be nice to have multiple raises so that we know where the issue is originating from. However, if we only want one raise, then I think it should be in `get_token` since there is an expectation that `get_token` returns a valid token.